website/source/docs/providers/fastly/index.html.markdown

```diff
@@ -0,0 +1,80 @@
+---
+layout: "fastly"
+page_title: "Provider: Fastly"
+sidebar_current: "docs-fastly-index"
+description: |-
+  Fastly
+---
+
+# Fastly Provider
+
+The Fastly provider is used to interact with the content delivery network (CDN) 
+provided by Fastly.
+
+In order to use this Provider, you must have an active account with Fastly.
+Pricing and signup information can be found at https://www.fastly.com/signup
+
+Use the navigation to the left to read about the available resources.
+
+## Example Usage
+
+```
+# Configure the Fastly Provider
+provider "fastly" {
+  api_key = "test"
+}
+
+# Create a Service  
+resource "fastly_service_v1" "myservice" {
+  name = "myawesometestservice"
+  ...
+}
+```
+
+## Authentication
+
+The Fastly provider offers an API key based method of providing credentials for
+authentication. The following methods are supported, in this order, and
+explained below:
+
+- Static API key
+- Environment variables
+
+
+### Static API Key ###
+
+Static credentials can be provided by adding a `api_key` in-line in the
+fastly provider block:
+
+Usage:
+
+```
+provider "fastly" {
+  api_key = "test"
+}
+```
+
+The API key for an account can be found on the Account page: https://app.fastly.com/#account
+
+###Environment variables
+
+You can provide your API key via `FASTLY_API_KEY` environment variable, 
+representing your Fastly API key.
+
+```
+provider "fastly" {}
```

Comment:
I think we should just mention that you can completely omit the `provider` block when the environment variable is specified, since that is the behavior, and I don't want to give the impression that the empty provider block is required.